### PR TITLE
Refactor realms handling

### PR DESCRIPTION
**Pull request by sadym-chromium**
* Move script handling to Realm.
* Move `ExecutionContextId` handling to Realm
* Introduced `BrowsingContextStorage`.
* Move `BrowsingContext` remove logic to `BrowsingContext`.
* Removed `browsingContextProcessor.spec.ts` in favour of e2e tests.

### CODE REVIEW

File: src/bidiMapper/domains/context/browsingContextStorage.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+export class BrowsingContextStorage {
+  static #contexts: Map<string, BrowsingContextImpl> = new Map();
+
+  static getTopLevelContexts(): BrowsingContextImpl[] {
+    return Array.from(BrowsingContextStorage.#contexts.values()).filter(
+      (c) => c.parentId === null
+    );
+  }
+
+  static forgetContext(contextId: string) {
```
Comment (thiagowfx) [inline src/bidiMapper/domains/context/browsingContextStorage.ts:30]: Can we call this `deleteContext`? `forget` is a bit non-standard w.r.t. CRUD terminology.

File: src/bidiMapper/domains/log/logManager.ts [hunk 2]
```diff
@@ -101,31 +102,38 @@ export class LogManager {
...
         await this.#bidiServer.sendMessage(
           new Log.LogEntryAddedEvent({
             level: 'error',
             source: {
-              // TODO sadym: add proper realm handling.
-              realm: (
-                params.exceptionDetails.executionContextId ?? 'UNKNOWN'
-              ).toString(),
-              context: this.#contextId,
+              realm: realm ? realm.realmId : 'UNKNOWN',
```
Comment (thiagowfx) [inline src/bidiMapper/domains/log/logManager.ts:135]: Optional simplification:

realm: realm?.realmId ?? 'UNKNOWN', ditto below

**Review by thiagowfx — APPROVED** [review]

File: src/bidiMapper/domains/context/browsingContextStorage.ts (hunk 1 above)
Comment (sadym-chromium) [reply src/bidiMapper/domains/context/browsingContextStorage.ts:30]: I wanted to underline it removes the context from storage.

Comment (sadym-chromium) [reply src/bidiMapper/domains/context/browsingContextStorage.ts]: Renamed `remove` to `delete` and `forget` to `remove`.

File: src/bidiMapper/domains/log/logManager.ts (hunk 2 above)
Comment (sadym-chromium) [reply src/bidiMapper/domains/log/logManager.ts]: done, thanks!

**Pull request merged**